### ISSUE TITLE
Gitlab Webhook for SSH repository

### ISSUE THREAD

**Issue by simplg** [WeblateOrg/weblate#8977]
### Describe the issue

I have both an on-premise gitlab server and weblate server. For security reasons, gitlab repositories are only accessible through ssh (for pulling and pushing) in a different port than the default 22. When testing the webhook for weblate, Gitlab receives an http code of 202 and weblate (in the logs) do not seem able to match the gitlab repository url with any project.

The following config for pulling and pushing projects in weblate is this url: `ssh://git@gitlab.example.com:4545/project/project`

The gitlab url (when accessing the repository webpage) is: `https://gitlab.example.com/project/project`
The webhook configured in gitlab repository webpage is: `https://weblate.example.com/hooks/gitlab/`

The response given in the logs by weblate when the webhook is triggered:
```
weblate-docker-weblate-1   | gunicorn stderr | [2023-03-23 09:49:16,660: INFO/376] received GitLab notification on repository project/project.git, URL https://gitlab.example.com/project/project, branch main, 0 matching components, 0 to process, 0 linked
```

I have also tried by adding in the ssh urls ".git" at the end, but I still got the same error. From what I've gathered by exploring different issues on this, it seems that the pulling url set in weblate has to match the one sent by the webhook. Since Gitlab sends the https url and I've configured weblate to use ssh, is there any way to make it work?

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

1. Create a new project on weblate using a private gitlab ssh repository using a different port than 22
2. Add a webhook to your gitlab repository pointing to your weblate server
3. Test it
4. Watch the logs through Docker `docker-compose logs -f`

### Expected behavior

It should match the repository with the correct project in weblate and update the local repository in weblate.

### How do you run Weblate?

Weblate on-premise

### Weblate versions

 * Weblate: 4.16.4
 * Django: 4.1.7
 * siphashc: 2.1
 * translate-toolkit: 3.8.6
 * lxml: 4.9.2
 * Pillow: 9.4.0
 * nh3: 0.2.8
 * python-dateutil: 2.8.2
 * social-auth-core: 4.4.0
 * social-auth-app-django: 5.1.0
 * django-crispy-forms: 2.0
 * oauthlib: 3.2.2
 * django-compressor: 4.3.1
 * djangorestframework: 3.14.0
 * django-filter: 22.1
 * django-appconf: 1.0.5
 * user-agents: 2.2.0
 * filelock: 3.10.0
 * rapidfuzz: 2.13.7
 * openpyxl: 3.1.2
 * celery: 5.2.7
 * django-celery-beat: 2.5.0
 * kombu: 5.2.4
 * translation-finder: 2.15
 * weblate-language-data: 2023.3
 * html2text: 2020.1.16
 * pycairo: 1.23.0
 * pygobject: 3.42.2
 * diff-match-patch: 20200713
 * requests: 2.28.2
 * django-redis: 5.2.0
 * hiredis: 2.2.2
 * sentry_sdk: 1.16.0
 * Cython: 0.29.33
 * misaka: 2.1.1
 * GitPython: 3.1.31
 * borgbackup: 1.2.3
 * pyparsing: 3.0.9
 * pyahocorasick: 2.0.0
 * python-redis-lock: 4.0.0
 * charset-normalizer: 3.1.0
 * Python: 3.11.2
 * Git: 2.30.2
 * psycopg2: 2.9.5
 * psycopg2-binary: 2.9.5
 * phply: 1.2.6
 * ruamel.yaml: 0.17.21
 * tesserocr: 2.6.0
 * boto3: 1.26.92
 * zeep: 4.2.1
 * aeidon: 1.12
 * iniparse: 0.5
 * mysqlclient: 2.1.1
 * Mercurial: 6.3.3
 * git-svn: 2.30.2
 * git-review: 2.3.1
 * Redis server: 7.0.10
 * PostgreSQL server: 15.2
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: smtp.example.com
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 5.4.0-132-generic (x86_64)

### Weblate deploy checks

```shell
System check identified some issues:

INFOS:
?: (weblate.I021) Error collection is not set up, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-4.16.4/admin/install.html#collecting-errors
?: (weblate.I028) Backups are not configured, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-4.16.4/admin/backup.html

System check identified 2 issues (1 silenced).
```

**Comment by nijel** [comment]
The matching is pretty flexible, it uses SSH URL provided by GitLab as well. I am not sure if your GitLab is configured to show the correct SSH URL, though.

If there is no exact match, only ending part of the repo is matched, but it turns out there is extra `.git` in it for GitLab. So having repository ending with `project/project.git` should do the trick for now. This was just fixed in a5dbbe94d8.

**Comment by simplg** [comment]
Thanks! It seems I didn't check enough and I can confirm that if I do add `.git` at the end, the hook seems to work as intended. 

Do you want me to close this issue or wait until a weblate version is released with this commit?

**Comment by nijel** [comment]
Closing as fixed for 4.17 release thanks to https://github.com/WeblateOrg/weblate/commit/a5dbbe94d81b7574c84d16e7abb593774a8a24ea
